### ISSUE TITLE
Connecting to IHaskell with ein kills the kernel

### ISSUE THREAD

**Issue by tonicebrian** [gibiansky/IHaskell#643]
I use [ein](https://github.com/tkf/emacs-ipython-notebook) for working and editing with notebooks in Emacs since I feel more productive than using the notebook's web interface. I tried use it for IHaskell but any time I connect the kernel dies.

The procedure to reproduce it is as follow:
- Start notebook using `stack ipython -- notebook`
- From the web interface create a new notebook
- Go to the notebook and change the kernel to Haskell. Until this point everything is smooth.
- Now open emacs and open the notebook with `M-x ein:notebooklist-open`, select the default port 8888, select the notebook we've just created, and then watch logs as the IHaskell kernel crashes. In the terminal you can see

```
stack exec ipython -- notebook                                                                                                              
Run from outside a project, using implicit global project config
Using resolver: lts-5.4 from implicit global project's config file: /home/cebrian/.stack/global-project/stack.yaml
[I 16:26:11.447 NotebookApp] Serving notebooks from local directory: /home/cebrian/GIT/thesis-nilm/notebooks
[I 16:26:11.447 NotebookApp] 0 active kernels 
[I 16:26:11.447 NotebookApp] The IPython Notebook is running at: http://localhost:8888/
[I 16:26:11.447 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
Se ha abierto una nueva ventana en la sesión actual del navegador.
[I 16:26:17.729 NotebookApp] Kernel started: 515346b9-fb93-4729-a9de-5736b15d2701
[I 16:26:22.210 NotebookApp] Kernel shutdown: 515346b9-fb93-4729-a9de-5736b15d2701
[I 16:26:22.228 NotebookApp] Kernel started: b1f274bb-b6b8-4e3e-b618-fa20a690e6f3
[W 16:26:22.253 NotebookApp] 404 GET /static/components/codemirror/mode/ihaskell/ihaskell.js?v=20160226162611 (127.0.0.1) 5.98ms referer=http://localhost:8888/notebooks/Untitled.ipynb
ihaskell: src/IHaskell/IPython/Message/Parser.hs:(63,5)-(68,51): Irrefutable pattern failed for pattern Success (messageType,
                                                               username,
                                                               messageUUID,
                                                               sessionUUID)

[I 16:27:01.228 NotebookApp] KernelRestarter: restarting kernel (1/5)
WARNING:root:kernel b1f274bb-b6b8-4e3e-b618-fa20a690e6f3 restarted
WARNING:root:kernel b1f274bb-b6b8-4e3e-b618-fa20a690e6f3 restarted
ihaskell: src/IHaskell/IPython/Message/Parser.hs:129:7-77: Irrefutable pattern failed for pattern Success (code,
                                                         silent,
                                                         storeHistory,
                                                         allowStdin)
```

and in the web interface the "Dead kernel" banner appears.

My setup is:
- Emacs 24.5
- ein-20160203.2026
- GHC 7.10.3
- IHaskell 0.8.3.0
- ipython 4.0.1
- python 2.7.6
- Running in an Ubuntu 14.04

It seems like is a problem of interaction between `ein` and `IHaskell` since I'm able to use the IHaskell from the web and a Python notebook from ein without problems. Maybe logging some information about what went wrong would be helpful.

Could you have a look and provide some guidelines about the required fix? I'd like to help fixing this issue since for me it is a stopper, so ask for any more info if required.

**Comment by zainab-ali** [comment]
This was the same error that I had.  It was caused by the `store_history` parameter being omitted from ein's message body.  This should be resolved by #757.

**Comment by vaibhavsagar** [comment]
Closing.
